src/wrappers/themis/themispp/secure_session.hpp

```diff
@@ -67,21 +70,14 @@ namespace themispp{
       _callback(NULL){
     }
 
+#if __cplusplus >= 201103L
+    DEPRECATED("use std::shared_ptr variant to transfer callback ownership instead")
```

Comment:
I might be wrong, but this deprecation message sounds complicated to understand for me (however I'm not a full time C++ dev, i don't know how typical deprecations looks like)

```suggestion
    DEPRECATED("call another constructor with std::shared_ptr to transfer callback ownership (secure_session_t(const data_t& id, const data_t& priv_key, std::shared_ptr<secure_session_callback_interface_t> &&callbacks))")
```

Reply:
Deprecation warnings typically look like this:

```
In file included from tests/themispp/main.cpp:20:
tests/themispp/secure_session_test.hpp:62:34: error: 'secure_session_t' is deprecated: use std::shared_ptr variant to transfer callback ownership instead [-Werror,-Wdeprecated-declarations]
      themispp::secure_session_t client(std::vector<uint8_t>(client_id.c_str(), client_id.c_str()+client_id.length()), std::vector<uint8_t>(client_priv, client_priv+sizeof(client_priv)), &client_callbacks);
                                 ^
src/wrappers/themis/themispp/secure_session.hpp:74:5: note: 'secure_session_t' has been explicitly marked deprecated here
    DEPRECATED("use std::shared_ptr variant to transfer callback ownership instead")
    ^
```

Now, after rereading this message I feel that it's not polite enough 😓 How about this one?

```suggestion
    DEPRECATED("please use std::shared_ptr<secure_session_callback_interface_t> constructor instead")
```

Reply:
yes, let's do your option